src/sentry/tasks/weekly_reports.py

```diff
@@ -0,0 +1,808 @@
+            ),
+            Condition(
+                Column("category"),
+                Op.IN,
+                [*DataCategory.error_categories(), DataCategory.TRANSACTION],
+            ),
+        ],
+        groupby=[Column("outcome"), Column("category"), Column("project_id"), Column("time")],
+        granularity=Granularity(ONE_DAY),
+        orderby=[OrderBy(Column("time"), Direction.ASC)],
+    )
+    request = Request(dataset=Dataset.Outcomes.value, app_id="reports", query=query)
+    data = raw_snql_query(request, referrer="weekly_reports.outcomes")["data"]
+
+    for dat in data:
+        project_id = dat["project_id"]
+        project_ctx = ctx.projects[project_id]
+        total = dat["total"]
+        timestamp = int(to_timestamp(parse_snuba_datetime(dat["time"])))
+        if dat["category"] == DataCategory.TRANSACTION:
+            # Transaction outcome
+            if dat["outcome"] == Outcome.RATE_LIMITED:
+                project_ctx.dropped_transaction_count += total
+            else:
+                project_ctx.accepted_transaction_count += total
+                project_ctx.transaction_count_by_day[timestamp] = total
+        else:
+            # Error outcome
+            if dat["outcome"] == Outcome.RATE_LIMITED:
+                project_ctx.dropped_error_count += total
+            else:
+                project_ctx.accepted_error_count += total
+                project_ctx.error_count_by_day[timestamp] = (
+                    project_ctx.error_count_by_day.get(timestamp, 0) + total
+                )
+
+
+def organization_project_issue_summaries(ctx):
+    all_issues = Group.objects.exclude(status=GroupStatus.IGNORED)
+    new_issue_counts = (
+        all_issues.filter(
+            project__organization_id=ctx.organization.id,
+            first_seen__gte=ctx.start,
+            first_seen__lt=ctx.end,
+        )
+        .values("project_id")
+        .annotate(total=Count("*"))
+    )
+    new_issue_counts = {item["project_id"]: item["total"] for item in new_issue_counts}
+
+    # Fetch all regressions. This is a little weird, since there's no way to
+    # tell *when* a group regressed using the Group model. Instead, we query
+    # all groups that have been seen in the last week and have ever regressed
+    # and query the Activity model to find out if they regressed within the
+    # past week. (In theory, the activity table *could* be used to answer this
+    # query without the subselect, but there's no suitable indexes to make it's
+    # performance predictable.)
+    reopened_issue_counts = (
+        Activity.objects.filter(
+            project__organization_id=ctx.organization.id,
+            group__in=all_issues.filter(
+                last_seen__gte=ctx.start,
+                last_seen__lt=ctx.end,
+                resolved_at__isnull=False,  # signals this has *ever* been resolved
+            ),
+            type__in=(ActivityType.SET_REGRESSION.value, ActivityType.SET_UNRESOLVED.value),
```

Comment:
This query was taken from the old weekly report generator. We're trying to count the number of reopened issues here. Reopened issues = Issues automatically reopened because we've detected the same errors again (SET_REGRESSION) + Issues manually reopened (SET_UNRESOLVED). Whenever one of these two events happen, we add a new row in the `Activity` table. So here we just need to count distinct `group_id` based on these criteria.